py4web/utils/auth.py

```diff
@@ -65,15 +65,18 @@ def abort_or_redirect(self, page, message=""):
         else redirects to page"""
         if re.search(REGEX_APPJSON, request.headers.get("accept", "")):
             abort(403)
-        redirect_next = request.fullpath
-        if request.query_string:
-            redirect_next = redirect_next + "?{}".format(request.query_string)
+        noappname_req_path = request.fullpath.split(f"/{request.app_name}")[-1]
```

Comment:
This will break tings if there is an action with the same name as the app_name

Reply:
Yes, there is the same issue now with _default app, no?

Reply:
My recollection is that we didn't need the 'next' fix if we have Val's bottle monkey patch fix.  Then next works fine.

Reply:
Yes, it is so.

Reply:
this is the opposite case where the application is served under subfolder (script_name header). In that case fullpath is script_name + path

Reply:
but ambiguous routes can still occur and as far as I understand this problem cannot be avoided

Reply:
the only way is to have strange names for apps like `_whatever` or `whatever_app` that are never used as action name

Reply:
For what its worth, the monkey patch solution is working really well for me in both dev and production environments.